# HG changeset patch
# User Ken Brown <kbrown@cornell.edu>
# Date 1425576628 18000
#      Thu Mar 05 12:30:28 2015 -0500
# Node ID 74b47369b904fba3b89b479173268d5382ab7189
# Parent  2130cb689a5d63ae7e97facbc2336c6517f64247
* src/lispbibl.d (ULONGLONG): Avoid conflict with w32 typedef on 64-bit Cygwin

diff -r 2130cb689a5d -r 74b47369b904 src/lispbibl.d
--- a/src/lispbibl.d	Thu Mar 05 11:35:30 2015 -0500
+++ b/src/lispbibl.d	Thu Mar 05 12:30:28 2015 -0500
@@ -1096,7 +1096,7 @@
 #else
   #error No 32 bit integer type? -- Which Integer-type has 32 Bit?
 #endif
-#if (long_bitsize==64)
+#if (long_bitsize==64) && !defined(UNIX_CYGWIN32)
   typedef long           SLONGLONG;
   typedef unsigned long  ULONGLONG;
   #ifndef HAVE_LONG_LONG_INT
@@ -1132,7 +1132,7 @@
 %%   emit_typedef("int","SLONG");
 %%   emit_typedef("unsigned int","ULONG");
 %% #endif
-%% #if (long_bitsize==64)
+%% #if (long_bitsize==64) && !defined(UNIX_CYGWIN32)
 %%   emit_typedef("long","SLONGLONG");
 %%   emit_typedef("unsigned long","ULONGLONG");
 %% #elif defined(MICROSOFT)
